packages/lit-html/src/lit-html.ts

```diff
@@ -929,13 +962,14 @@ export class AttributePart {
       // TODO (justinfagnani): To support nested directives, we'd need to
       // resolve the directive result's values. We may want to offer another
       // way of composing directives.
-      value = directive.update(this, (value as DirectiveResult).values);
+      value = directive.resolve(this, (value as DirectiveResult).values);
     }
-    return value ?? '';
+    return value;
```

Comment:
This was coercing a directive that returned `undefined` or `null` to a string; however that coercion should only be done on the final value of an attribute part prior to commit only for AttributePart, not PropertyPart, since PropertyParts want to pass `undefined`/`null` to their binding.  It was just being in the wrong spot, highlighted by the SSR tests. I added `lit-html` tests for this: https://github.com/Polymer/lit-html/pull/1347/files/a4255d80d26221d928c3e1b7daa3e78148fdb4e1#diff-652423ad0006947a62a8c6714793599f92da50a1cfeb758ff314897790d9ddafR1175-R1208

Reply:
got it.